src/python/pants/backend/go/goals/test.py

```diff
@@ -222,8 +243,16 @@ def compilation_failure(exit_code: int, stdout: str | None, stderr: str | None)
         )
     test_pkg_build_request = maybe_test_pkg_build_request.request
 
+    # TODO: Eventually support adding coverage to non-test packages. Those other packages will need to be
+    # added to `main_direct_deps` and to the coverage setup in the testmain.
     main_direct_deps = [test_pkg_build_request]
 
+    # Build the `main_direct_deps` when in coverage mode so we can obtain coverage variables.
+    # TODO: Consider a way to generate coverage variables without building.
```

Comment:
It's not clear to me what this means... might be good to expand on the benefits.

Reply:
Will do. For context, Go coverage analysis is a form of codegen. It rewrites the Go source code being examined to include explicit references to "coverage variables" which contain the statement counts for coverage analysis. The test main generated for a Go test binary has to explicitly reference the coverage variables generated by this codegen and register them with the coverage runtime.

Currently, I have the names of the coverage variables output as a result of building the package under test. Those variables could probably be output as a result of just running the code generation. In the end, it probably doesn't matter since the package under test will end up being built any way and Pants will cache that build so making it a dependency of the test main (as needs to happen) won't result in duplicative work.